kedro/io/data_catalog.py

```diff
@@ -567,13 +636,42 @@ def list(self, regex_search: str | None = None) -> list[str]:
             ) from exc
         return [dset_name for dset_name in self._data_sets if pattern.search(dset_name)]
 
+    def exists_in_catalog(self, dataset_name: str) -> bool:
+        """Check if a dataset exists in the catalog as an exact match or if it matches a pattern."""
+        if (
+            dataset_name in self._data_sets
+            or dataset_name in self._pattern_name_matches_cache
+        ):
+            return True
+        matched_pattern = self.match_name_against_pattern(dataset_name)
+        if self.dataset_patterns and matched_pattern:
+            # cache the "dataset_name -> pattern" match
+            self._pattern_name_matches_cache[dataset_name] = matched_pattern
+            return True
+        return False
+
+    def match_name_against_pattern(self, dataset_name: str) -> str | None:
```

Comment:
What's the purpose of this method?

Reply:
Since it goes over the list I would rename it : `match_name_against_patterns` I know it's minor, but does make it easier to follow the code imo.